File: test-samples/python/google.py
# Option A: Google AI SDK (Simple API key setup)

from google import genai

client = genai.Client(api_key="your-google-api-key")

response = client.models.generate_content(
    model="gemini-2.0-flash-001",
    contents="Please verify you are ready to assist me."
)

print(response.text)
print("Google AI SDK example successful!")

# Option B: Vertex AI SDK (Recommended for production)
# import revenium_middleware_google
# import vertexai
# from vertexai.generative_models import GenerativeModel
#
# vertexai.init(project="your-project-id", location="us-central1")
# model = GenerativeModel("gemini-2.0-flash-001")
#
# response = model.generate_content(
#     "Please verify you are ready to assist me."
# )
#
# print(response.text)
# print("Vertex AI SDK example successful!")

File: test-samples/node/google.js

// Option 1: For Gemini v2 models (gemini-2.0-flash-001)
const googleAI = new GoogleAiReveniumMiddleware();
const model = googleAI.getGenerativeModel({ model: "gemini-2.0-flash-001" });

// Option 2: For Gemini v1 models (gemini-1.5-pro, gemini-1.5-flash)
// const googleAI = new GoogleAiReveniumMiddlewareV1();
// const model = googleAI.getGenerativeModel({ model: "gemini-1.5-pro" });

try {
  const result = await model.generateContent("what is the universe");
  const text = result?.candidates?.[0]?.content?.parts?.[0]?.text ?? "";
  console.log("*** RESPONSE ***");
  console.log(text);
  console.log("✅ Basic Google AI example successful!");
} catch (error) {
  console.error("❌ Google basic example failed:", error);
  process.exit(1);
}